src/Discord.Net.Commands/Attributes/Preconditions/RequirePermission.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Discord.Commands.Attributes.Preconditions
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public class RequirePermission : RequireGuildAttribute
+    {
+        public GuildPermission? GuildPermission { get; set; }
+        public ChannelPermission? ChannelPermission { get; set; }
+
```

Comment:
I decided to leave it this way for developers who wanted to define their permission using both; something like this:

```cs
[RequirePermission(GuildPermission = GuildPermission.ABC, ChannelPermission = ChannelPermission.DEF)]
```

Having a public setter allows properties to be set by name; like the AllowMultiple and Inherited properties in AttributeUsage.